spec/classes/defaults_spec.rb

```diff
@@ -63,20 +63,35 @@
         is_expected.to contain_logrotate__conf('/etc/logrotate.conf')
       }
 
-      it {
-        is_expected.to contain_logrotate__rule('wtmp').with(
-          'path' => '/var/log/wtmp',
-          'create_mode' => '0664',
-          'missingok' => false,
-          'minsize' => '1M',
-          'create' => true,
-          'create_owner' => 'root',
-          'create_group' => 'utmp',
-          'rotate' => 1,
-          'rotate_every' => 'monthly'
-        )
-      }
-
+      if facts[:operatingsystemmajrelease].to_i <= 7
+        it {
+          is_expected.to contain_logrotate__rule('wtmp').with(
+            'path' => '/var/log/wtmp',
+            'create_mode' => '0664',
+            'missingok' => false,
+            'minsize' => '1M',
+            'create' => true,
+            'create_owner' => 'root',
+            'create_group' => 'utmp',
+            'rotate' => 1,
+            'rotate_every' => 'monthly'
+          )
+        }
+      else
+        it {
+          is_expected.to contain_logrotate__rule('wtmp').with(
+            'path' => '/var/log/wtmp',
+            'create_mode' => '0664',
+            'missingok' => true,
+            'minsize' => '1M',
+            'create' => true,
+            'create_owner' => 'root',
+            'create_group' => 'utmp',
+            'rotate' => 1,
+            'rotate_every' => 'monthly'
+          )
+        }
+      end
```

Comment:
Rather than duplicating it, thoughts on the following (where I didn't bother with correctly indenting):
```suggestion
        it do
          is_expected.to contain_logrotate__rule('wtmp').with(
            'path' => '/var/log/wtmp',
            'create_mode' => '0664',
            'missingok' => facts[:operatingsystemmajrelease].to_i >= 8,
            'minsize' => '1M',
            'create' => true,
            'create_owner' => 'root',
            'create_group' => 'utmp',
            'rotate' => 1,
            'rotate_every' => 'monthly'
          )
      end
```

Reply:
done